### PR TITLE
fix for mac (error loading lib)

### PR DESCRIPTION
**Pull request by Barthelemy**
Also a bit of cleanup in the code

### CODE REVIEW

**Comment by Barthelemy** [comment]
I should have kept the code refactoring apart but I guess it is ok to have it in this PR.

File: Framework/include/QualityControl/CheckerDataProcessor.h [hunk 1]
```diff
@@ -35,71 +35,94 @@ namespace checker {
 /// TODO Evaluate whether we should have a dedicated device to store in the database.
 ///
 /// \author Barthélémy von Haller
-class CheckerDataProcessor : public framework::Task {
```
Comment (knopers8) [inline Framework/include/QualityControl/CheckerDataProcessor.h]: There is actually discrepancy between clang-format in O2 and QualityControl (i think QC is closer to Coding Guidelines). Maybe you could mention that in WP3?

File: Framework/src/CheckerDataProcessor.cxx [hunk 2]
```diff
@@ -31,6 +31,7 @@ using namespace repository;
 namespace checker {
 
 // TODO do we need a CheckFactory ? here it is embedded in the Checker
+// TODO maybe we could use the CheckerDataProcessorFactory
```
Comment (knopers8) [inline Framework/src/CheckerDataProcessor.cxx]: I am not sure if I understand - `CheckerDataProcessorFactory` for both checkers and checks?

File: Framework/src/CheckerDataProcessor.cxx [hunk 3]
```diff
@@ -62,7 +63,9 @@ void CheckerDataProcessor::init(framework::InitContext&)
   // configuration
   try {
     std::unique_ptr<ConfigurationInterface> config = ConfigurationFactory::getConfiguration(mConfigurationSource);
-    populateConfig(config);
+    // configuration of the database
+    mDatabase = DatabaseFactory::create(config->get<std::string>("qc/config/database/implementation").value());
+    mDatabase->connect(config);
```
Comment (knopers8) [inline Framework/src/CheckerDataProcessor.cxx]: Thank you, it has more sense now.

File: Framework/include/QualityControl/CheckerDataProcessor.h (hunk 1 above)
Comment (Barthelemy) [reply Framework/include/QualityControl/CheckerDataProcessor.h]: I will, thank you.

File: Framework/src/CheckerDataProcessor.cxx (hunk 2 above)
Comment (Barthelemy) [reply Framework/src/CheckerDataProcessor.cxx]: yes, with a rename :)

**Pull request merged**